src/hotspot/cpu/aarch64/templateTable_aarch64.cpp

```diff
@@ -2357,32 +2366,111 @@ void TemplateTable::load_resolved_field_entry(Register obj,
 
-  // klass overwrite register
-  if (is_static) {
-    __ ldr(obj, Address(cache, in_bytes(cp_base_offset +
-                                        ConstantPoolCacheEntry::f1_offset())));
-    const int mirror_offset = in_bytes(Klass::java_mirror_offset());
-    __ ldr(obj, Address(obj, mirror_offset));
-    __ resolve_oop_handle(obj, r5, rscratch2);
-  }
+  // determine constant pool cache field offsets
+  resolve_cache_and_index_for_method(f1_byte, cache, index);
+  __ load_unsigned_byte(flags, Address(cache, in_bytes(ResolvedMethodEntry::flags_offset())));
+  __ ldr(method, Address(cache, in_bytes(ResolvedMethodEntry::method_offset())));
+}
+
+void TemplateTable::load_resolved_method_entry_handle(Register cache,
+                                                      Register method,
+                                                      Register ref_index,
+                                                      Register flags) {
+  // setup registers
+  //const Register cache = rscratch2;
+  const Register index = r4;
+  assert_different_registers(method, flags);
+  assert_different_registers(method, cache, index);
+
+  // determine constant pool cache field offsets
+  resolve_cache_and_index_for_method(f1_byte, cache, index);
+  __ load_unsigned_byte(flags, Address(cache, in_bytes(ResolvedMethodEntry::flags_offset())));
+
+  // invokehandle uses an index into the resolved references array
+  __ ldr(method, Address(cache, in_bytes(ResolvedMethodEntry::method_offset())));
+  __ load_unsigned_short(ref_index, Address(cache, in_bytes(ResolvedMethodEntry::resolved_references_index_offset())));
+
+  // maybe push appendix to arguments (just before return address)
+  Label L_no_push;
+  __ tbz(flags, ResolvedMethodEntry::has_appendix_shift, L_no_push);
+  // Push the appendix as a trailing parameter.
+  // This must be done before we get the receiver,
+  // since the parameter_size includes it.
+  __ push(r19);
+  __ mov(r19, ref_index);
+  __ load_resolved_reference_at_index(ref_index, r19);
+  __ pop(r19);
+  __ push(ref_index);  // push appendix (MethodType, CallSite, etc.)
+  __ bind(L_no_push);
+}
+
+void TemplateTable::load_resolved_method_entry_interface(Register cache,
+                                                         Register klass,
+                                                         Register method_or_table_index,
+                                                         Register flags) {
+  // setup registers
+  const Register index = r4;
+  assert_different_registers(method_or_table_index, flags);
+  assert_different_registers(method_or_table_index, cache, flags);
+
+  // determine constant pool cache field offsets
+  resolve_cache_and_index_for_method(f1_byte, cache, index);
+  __ load_unsigned_byte(flags, Address(cache, in_bytes(ResolvedMethodEntry::flags_offset())));
+
+  // table_or_ref_index can either be an itable index or a resolved reference index depending on the bytecode
```

Comment:
This comment seems to apply to `ref_index` in `load_resolved_method_entry_handle`? Does it need to move up there? Do we need a comment here stating somethgin similar for `method_or_table_index`?

Reply:
This looks like a leftover from a previous implementation. I don't believe this is true for either invokeinterface or invokehandle.